src/collections/config/schema.ts

```diff
@@ -89,7 +89,8 @@ const collectionSchema = joi.object().keys({
       maxLoginAttempts: joi.number(),
       disableLocalStrategy: joi.boolean().valid(true),
       strategies: joi.array().items(joi.object().keys({
-        strategy: joi.object().required(),
+        name: joi.string().required(),
+        strategy: joi.func().maxArity(1).required(),
```

Comment:
We can do some joi.alternatives to either have the function pattern you made or allow the object as before.